FUSETOOLS-2018 - updates to work better with Fuse servers

bfitzpat opened the pull request:
This PR handles a few things:
1) It removes the duplicate declarations for Fuse server types. Those are provided by Fuseide and no longer needed here since we're not installing SY separately from the Fuse tooling.
2) We have fixed the default Java version finally to 1.7 instead of 1.6. This was an oversight on my part.
3) We now refresh the project and do a Maven->Update configuration when we create a new project, which generates the MANIFEST.MF out of the box (it refreshes when the project is updated after that). This ensures that users can deploy their projects to a Fuse server without incident.

bfitzpat commented:
A big thank you goes to @lhein  and Fred Bricon for their help on this.

lhein commented:
+1 after successful testing / building of the PR job

apupier commented on eclipse/plugins/org.switchyard.tools.ui/src/org/switchyard/tools/ui/operations/CreateSwitchYardProjectOperation.java:
```diff
@@ -484,6 +486,20 @@ public void run(IProgressMonitor monitor) throws CoreException {
+        // update the project so we ensure a Project->Clean is done so the
+        // MANIFEST.MF is built and we don't run into trouble deploying the 
+        // project on a Fuse server.
+        if (_projectMetatData.getNewProjectHandle() != null) {
```
_projectMetatData.getNewProjectHandle() can be extracted as local variable as it is used 3 times

apupier commented:
Here is test in failure:

-------------------------------------------------------------------------------
Test set: org.switchyard.tools.ui.tests.SwitchYardValidatorTest
-------------------------------------------------------------------------------
Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 137.841 sec <<< FAILURE! - in org.switchyard.tools.ui.tests.SwitchYardValidatorTest
testWiringValidation(org.switchyard.tools.ui.tests.SwitchYardValidatorTest)  Time elapsed: 114.583 sec  <<< FAILURE!
junit.framework.AssertionFailedError: Expecting 6 warnings: Type=org.switchyard.tools.ui.problemMarker:Message=Duplicate Name: The name "ExtendedService" must be unique among services, references and component services.:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=Ambiguous Reference: Cannot resolve a service for component reference "ExtendedService" (i.e. resolves to multiple services).:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=Duplicate Name: The name "ExtendedService" must be unique among services, references and component services.:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=No interface is defined for the service/reference.:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=Duplicate Name: The name "ExtendedService" must be unique among services, references and component services.:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=Unresolved Reference: The component reference "UnresolvedReference" cannot be resolved to a service.:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=Service Interface Mismatch: Interface for service "WSDLService" in component "InvalidInterfaceComponent" does not match the interface provided by the implementation.:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=Unused Component Reference: Reference "WSDLReference" defined by component "InvalidInterfaceComponent" is not used by the component's implementation.:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=Incompatible Interface Type: "WSDL" interfaces cannot be used with "Bean" components (InvalidInterfaceComponent/WSDLService).:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=Unresolved Reference: The component reference "WSDLReference" cannot be resolved to a service.:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=Incompatible Interface Type: "WSDL" interfaces cannot be used with "Bean" components (InvalidInterfaceComponent/WSDLReference).:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=Required Capability Missing: The "File Transfer (FTP/FTPS/SFTP)" capability should be configured on the project when using this type of binding.:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=Unused Capability: The "File" capability is configured on the project, but is not required by any SwitchYard binding or implementation types used in the project.:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=Error loading service interface meta-data.  Interface compatibility cannot be verified.:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=Required transformation missing: from="{urn:org.switchyard.tools.tests:wiring-validation-tests:0.0.1-SNAPSHOT}testInOnlyMethod" to="java:java.lang.String".:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=Required transformation missing: from="java:int" to="{urn:org.switchyard.tools.tests:wiring-validation-tests:0.0.1-SNAPSHOT}testInOutMethodResponse".:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=Unbound Operation: Cannot resolve operation "testMissingOperation" on target.:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=Unbound Service: The service "UnboundService" is not bound to any component service (i.e. it promotes no component service).:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=The referenced Java interface for service "UnboundService" specifies more than one operation and the binding "null" does not specify which to use in the operation selector.:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=No interface is defined for the service/reference.:LineNumber=null, Type=org.switchyard.tools.ui.problemMarker:Message=Unused Reference: The reference "UnusedReference" is unused (i.e. it promotes no component references).:LineNumber=null expected:<6> but was:<5>
	at junit.framework.Assert.fail(Assert.java:57)
	at junit.framework.Assert.failNotEquals(Assert.java:329)
	at junit.framework.Assert.assertEquals(Assert.java:78)
	at junit.framework.Assert.assertEquals(Assert.java:234)
	at junit.framework.TestCase.assertEquals(TestCase.java:401)
	at org.switchyard.tools.ui.tests.SwitchYardValidatorTest.testWiringValidation(SwitchYardValidatorTest.java:72)

testCamelRouteValidation(org.switchyard.tools.ui.tests.SwitchYardValidatorTest)  Time elapsed: 23.256 sec

apupier commented:
The missing validation is:
Type=org.switchyard.tools.ui.problemMarker:Message=Error loading service interface meta-data.  Interface compatibility cannot be verified.:LineNumber=null,

Do you see an explanation? maybe it is an issue fixed by side effect?

apupier commented:
if I change the check to expect only 5 warning, the test is then failing on:

`junit.framework.AssertionFailedError: Type=org.eclipse.wst.xml.core.validationMarker:Message=No grammar constraints (DTD or XML Schema) referenced in the document.:LineNumber=1, Type=org.eclipse.wst.xml.core.validationMarker:Message=cvc-complex-type.4: Attribute 'promote' must appear on element 'sca:service'.:LineNumber=62, Type=org.eclipse.wst.xml.core.validationMarker:Message=cvc-complex-type.4: Attribute 'promote' must appear on element 'sca:reference'.:LineNumber=74, Type=org.eclipse.wst.xml.core.validationMarker:Message=cvc-complex-type.4: Attribute 'promote' must appear on element 'sca:reference'.:LineNumber=87, Type=org.eclipse.wst.xml.core.validationMarker:Message=cvc-complex-type.2.4.a: Invalid content was found starting with element 'sca:reference'. One of '{"http://docs.oasis-open.org/ns/opencsa/sca/200912":documentation, "http://docs.oasis-open.org/ns/opencsa/sca/200912":implementation}' is expected.:LineNumber=14, Type=org.eclipse.wst.xml.core.validationMarker:Message=cvc-complex-type.2.4.a: Invalid content was found starting with element 'sca:reference'. One of '{"http://docs.oasis-open.org/ns/opencsa/sca/200912":documentation, "http://docs.oasis-open.org/ns/opencsa/sca/200912":implementation}' is expected.:LineNumber=22, Type=org.eclipse.wst.xml.core.validationMarker:Message=cvc-complex-type.2.4.a: Invalid content was found starting with element 'sca:service'. One of '{"http://docs.oasis-open.org/ns/opencsa/sca/200912":documentation, "http://docs.oasis-open.org/ns/opencsa/sca/200912":implementation}' is expected.:LineNumber=30, Type=org.eclipse.wst.xml.core.validationMarker:Message=cvc-complex-type.2.4.a: Invalid content was found starting with element 'sca:reference'. One of '{"http://docs.oasis-open.org/ns/opencsa/sca/200912":documentation, "http://docs.oasis-open.org/ns/opencsa/sca/200912":implementation}' is expected.:LineNumber=38, Type=org.eclipse.wst.xml.core.validationMarker:Message=cvc-complex-type.2.4.a: Invalid content was found starting with element 'sca:service'. One of '{"http://docs.oasis-open.org/ns/opencsa/sca/200912":documentation, "http://docs.oasis-open.org/ns/opencsa/sca/200912":implementation}' is expected.:LineNumber=44, Type=org.eclipse.wst.xml.core.validationMarker:Message=cvc-complex-type.2.4.a: Invalid content was found starting with element 'sca:reference'. One of '{"http://docs.oasis-open.org/ns/opencsa/sca/200912":documentation, "http://docs.oasis-open.org/ns/opencsa/sca/200912":implementation}' is expected.:LineNumber=52 expected:<0> but was:<10>
	at junit.framework.Assert.fail(Assert.java:57)
	at junit.framework.Assert.failNotEquals(Assert.java:329)
	at junit.framework.Assert.assertEquals(Assert.java:78)
	at junit.framework.Assert.assertEquals(Assert.java:234)
	at junit.framework.TestCase.assertEquals(TestCase.java:401)
	at org.switchyard.tools.ui.tests.SwitchYardValidatorTest.testWiringValidation(SwitchYardValidatorTest.java:119)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at junit.framework.TestCase.runTest(TestCase.java:176)
	at org.eclipse.m2e.tests.common.AbstractMavenProjectTestCase.runTest(AbstractMavenProjectTestCase.java:576)
	at junit.framework.TestCase.runBare(TestCase.java:141)
	at junit.framework.TestResult$1.protect(TestResult.java:122)
	at junit.framework.TestResult.runProtected(TestResult.java:142)
	at junit.framework.TestResult.run(TestResult.java:125)
	at junit.framework.TestCase.run(TestCase.java:129)
	at junit.framework.TestSuite.runTest(TestSuite.java:252)
	at junit.framework.TestSuite.run(TestSuite.java:247)
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.pde.internal.junit.runtime.RemotePluginTestRunner.main(RemotePluginTestRunner.java:62)
	at org.eclipse.pde.internal.junit.runtime.PlatformUITestHarness$1.run(PlatformUITestHarness.java:47)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:35)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:135)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4155)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3772)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:827)
	at org.eclipse.jface.window.Window.open(Window.java:803)
	at org.jboss.tools.usage.event.UsageReporter$6.run(UsageReporter.java:364)
	at org.eclipse.ui.internal.UILockListener.doPendingWork(UILockListener.java:162)
	at org.eclipse.ui.internal.UISynchronizer$3.run(UISynchronizer.java:154)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:35)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:135)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4155)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3772)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:1127)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:337)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1018)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:156)
	at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:694)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:337)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:606)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:150)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:139)
	at org.eclipse.pde.internal.junit.runtime.NonUIThreadTestApplication.runApp(NonUIThreadTestApplication.java:54)
	at org.eclipse.pde.internal.junit.runtime.UITestApplication.runApp(UITestApplication.java:47)
	at org.eclipse.pde.internal.junit.runtime.NonUIThreadTestApplication.start(NonUIThreadTestApplication.java:48)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:380)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:235)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:669)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:608)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1515)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1488)

`

apupier commented:
@bfitzpat I provided a PR removing the check for missing validation (in case this is really normal): https://github.com/bfitzpat/tools/pull/1

bfitzpat commented:
It's not normal. Tests pass fine here and I think this is just a funky thing in Jenkins again.

bfitzpat commented:
This should not be failing where it is.

bfitzpat commented:
Updated the PR to address Aurelien's comment about repeated code and updated the JBTIS TP version to see if that has any effect to shake this loose. I am unable to reproduce this locally with the same test and same test project.

apupier commented:
> It's not normal. Tests pass fine here and I think this is just a funky thing in Jenkins again.

It was failing on my machine exactly as on CI.

What is the goal of the following validation marker?
Type=org.switchyard.tools.ui.problemMarker:Message=Error loading service interface meta-data. Interface compatibility cannot be verified.:LineNumber=null,

bfitzpat commented:
https://github.com/jboss-switchyard/tools/blob/master/eclipse/plugins/org.switchyard.tools.ui/src/org/switchyard/tools/ui/validation/WiringValidationContext.java#L242

We have to ensure that we can get to the classes that are referenced by the service interfaces used in the SwitchYard config. It's a classloader thing. So if we can't find them, that's a problem.

bfitzpat commented:
Ok. Updated a couple of things - we had a warning that disappeared as Aurelien pointed out. I suspect it was due to some fixes elsewhere. And the Java version in the camel test project needed to be updated - it was causing an issue. It's now 1.7 to match the default version of the tooling. Waiting for PR build to make sure, but the failing tests now pass locally after I managed to reproduce the issues.

bfitzpat commented:
Updated the PR to remove the refreshLocal causing more frequent test failures
